### PR TITLE
add prometheus-reporter sidecar

### PR DESCRIPTION
**Pull request by sidler-mozilla**

### CODE REVIEW

File: charts/refractr/templates/ingress-hook.yaml [hunk 1]
```diff
@@ -71,36 +71,6 @@ subjects:
     name: refractr
 ---
 
-apiVersion: helm.fluxcd.io/v1
```
Comment (The-smooth-operator) [inline charts/refractr/templates/ingress-hook.yaml:74]: Do you know why there is a HelmRelease inside a Helm Chart? Is this needed or is a bug?

If we need this, it has to be added to the Chart.yaml file as a dependency

File: charts/refractr/Chart.yaml [hunk 2]
```diff
@@ -17,3 +17,9 @@ sources:
 maintainers:
   - name: Alberto del Barrio
     email: alberto@mozilla.com
+
+dependencies:
+- name: cert-manager
+  version: v0.15.2
+  contidion: cert-manager.install
```
Comment (mozafrank) [inline charts/refractr/Chart.yaml:24]: typo?

File: charts/refractr/values.yaml [hunk 3]
```diff
@@ -87,3 +95,81 @@ nginx-ingress:
...
+    ##
+    # pullSecrets:
+    #   - myRegistryKeySecretName
+
+  ## Prometheus exporter pods' annotation and labels
+  ## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
+  ##
+  podAnnotations:
+    prometheus.io/scrape: "true"
+    prometheus.io/port: "9113"
```
Comment (mozafrank) [inline charts/refractr/values.yaml:124]: Should this be {{ .Values.metrics.service.port }} ?

File: charts/refractr/templates/ingress-hook.yaml (hunk 1 above)
Comment (The-smooth-operator) [reply charts/refractr/templates/ingress-hook.yaml:74]: This was solved moving it to a dependency.

File: charts/refractr/Chart.yaml (hunk 2 above)
Comment (The-smooth-operator) [reply charts/refractr/Chart.yaml:24]: yes, thanks for catching this one

File: charts/refractr/values.yaml (hunk 3 above)
Comment (The-smooth-operator) [reply charts/refractr/values.yaml:124]: mmmmmm I'm not sure if that is a good choice actually. 
Choosing here what seems to be a random port number might not be the best option, but I believe Prometheus will autodiscover it.
The problem adding there the same port that used in the service definition is that probably those metrics can be fetched externally (in the internet) by hitting the endpoint URL/metrics.
I guess we need to test it

**Pull request merged**